src/cdk/testing/testbed/unit-test-element.ts

```diff
@@ -78,15 +78,19 @@ export class UnitTestElement implements TestElement {
     await this._stabilize();
   }
 
-  async click(...args: number[]): Promise<void> {
-    const {left, top, width, height} = await this.getDimensions();
-    const relativeX = args.length ? args[0] : width / 2;
-    const relativeY = args.length ? args[1] : height / 2;
-
-    // Round the computed click position as decimal pixels are not
-    // supported by mouse events and could lead to unexpected results.
-    const clientX = Math.round(left + relativeX);
-    const clientY = Math.round(top + relativeY);
+  async click(...args: ['center'] | [number, number]): Promise<void> {
+    let clientX: number | undefined = undefined;
+    let clientY: number | undefined = undefined;
+    if (args.length) {
+      const {left, top, width, height} = await this.getDimensions();
```

Comment:
It'll be a bit longer, but you can avoid the `getDimensions` call if the coordinates are passed in. E.g.

```
let relativeX: number | undefined;
let relativeY: number | undefined;

if (args[0] === 'center') {
  const {left, top, width, height} = await this.getDimensions();
  ...
} else {
  relativeX = args[0];
  relativeY = args[1];
}
```